src/libappimage/libappimage.cpp

```diff
@@ -0,0 +1,364 @@
+
+                    } else nextHoop = itr.linkTarget();
+
+                    break;
+                }
+            }
+
+
+            if (!nextHoop.empty())
+                targetFile = nextHoop;
+        } catch (const std::runtime_error& err) {
+            std::clog << "Error at " << __FUNCTION__ << " : " << err.what() << std::endl;
+        } catch (...) {
+            std::clog << "Error at " << __FUNCTION__ << " : that's all we know." << std::endl;
+        }
+    }
+    return false;
+}
+
+void appimage_extract_file_following_symlinks(const char* appimage_file_path, const char* file_path,
+                                              const char* target_file_path) {
+
+    std::string targetFile = file_path;
+    std::vector<std::string> visitedEntries;
+
+    while (!targetFile.empty()) {
+        // Find loops
+        if (std::find(visitedEntries.begin(), visitedEntries.end(), targetFile) != visitedEntries.end()) {
+            std::clog << "WARNING: Links loop found while extracting " << file_path;
+            break;
+        }
+
+        visitedEntries.emplace_back(targetFile);
+        std::string nextHoop;
+
+        try {
+            AppImage appImage(appimage_file_path);
+
+            for (auto itr = appImage.files(); itr != itr.end(); ++itr)
+                if (itr.path() == targetFile) {
+                    if (itr.type() == PayloadEntryType::REGULAR) {
+                        bf::ofstream output(target_file_path);
+                        output << itr.read().rdbuf();
+
+                        return;
+                    } else nextHoop = itr.linkTarget();
+
+                    break;
+                }
+        } catch (const std::runtime_error& err) {
+            std::clog << "ERROR: " << __FUNCTION__ << " : " << err.what() << std::endl;
+        } catch (...) {
+            std::clog << "ERROR: " << __FUNCTION__ << " failed. That's all we know." << std::endl;
+        }
+
+        targetFile = nextHoop;
+    }
+}
+
+
+/*
+ * Checks whether an AppImage's desktop file has set X-AppImage-Integrate=false.
+ * Useful to check whether the author of an AppImage doesn't want it to be integrated.
+ *
+ * Returns >0 if set, 0 if not set, <0 on errors.
+ */
+int appimage_shall_not_be_integrated(const char* path) {
+    try {
+        AppImage appImage(path);
+        XdgUtils::DesktopEntry::DesktopEntry entry;
+        // Load Desktop Entry
+        for (auto itr = appImage.files(); itr != itr.end(); ++itr) {
+            const auto& entryPath = *itr;
+            if (entryPath.find(".desktop") != std::string::npos && entryPath.find('/') == std::string::npos) {
+                itr.read() >> entry;
+                break;
+            }
+        }
+
+        auto integrateEntryValue = entry.get("Desktop Entry/X-AppImage-Integrate", "true");
+
+        boost::to_lower(integrateEntryValue);
+        boost::algorithm::trim(integrateEntryValue);
+
+        return integrateEntryValue == "false";
```

Comment:
This code should not have to be in here, your library shall specification details like "interpretation of boolean values". ld-desktopfile can do it (it's conformance tested even).

Reply:
Totally valid improvement. But not critical, will take a look at it once done with the review.

Reply:
Anything you want to defer should go into an issue.